app/models/git_repository.rb

```diff
@@ -84,7 +84,7 @@ def update_repo
     with_worktree do |worktree|
       message = "Updating #{url} in #{directory_name}..."
       _log.info(message)
-      worktree.send(:fetch_and_merge)
+      worktree.send(:pull)
```

Comment:
This is a bug waiting to happen, since pull locks and fetch_and_merge doesn't.  I can remove from this PR if needed.

Reply:
Made it a separate commit